Engine/Code/Engine/Graphics/Renderer.cpp

```diff
@@ -335,12 +382,27 @@ void Renderer::DrawFrameBuffer(const Vec2& position, Vec2 meshDim)
 {
 	MeshBuilder* mb = new MeshBuilder();
 
-	mb->Push(VertexMaster(Vec3(position.m_X              , position.m_Y              , 0.0f), Vec4(1.0f, 1.0f, 1.0f, 1.0f), Vec2(0.0f, 0.0f)));
-	mb->Push(VertexMaster(Vec3(position.m_X + meshDim.m_X, position.m_Y              , 0.0f), Vec4(1.0f, 1.0f, 1.0f, 1.0f), Vec2(1.0f, 0.0f)));
-	mb->Push(VertexMaster(Vec3(position.m_X + meshDim.m_X, position.m_Y + meshDim.m_Y, 0.0f), Vec4(1.0f, 1.0f, 1.0f, 1.0f), Vec2(1.0f, 1.0f)));
-	mb->Push(VertexMaster(Vec3(position.m_X              , position.m_Y + meshDim.m_Y, 0.0f), Vec4(1.0f, 1.0f, 1.0f, 1.0f), Vec2(0.0f, 1.0f)));
+	mb->Color3f(Vec4(1.0f, 1.0f, 1.0f, 1.0f));
+	mb->TexCoord2f(Vec2(0.0f, 0.0f));
+	mb->Position3f(Vec3(position.m_X, position.m_Y, 0.0f));
+	mb->Push();
+
+	mb->Color3f(Vec4(1.0f, 1.0f, 1.0f, 1.0f));
```

Comment:
Whole reason for doing this is if the color doesn't change you don't have to keep using that call. 
mb->Color3f(White)
mb->Pos3f(...)
mb->push
mb->Pos3f(...)
mb->Push
mb->Pos3f(..)

All of these while be white since that color is cached. Likewise with texcoords or any other value thats added.